### PR TITLE
Merge Joe's changes for boto3 to develop

### PR DESCRIPTION
**Pull request by ProfOak**

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,5 +1,6 @@
 xmltodict==0.9.2
 boto>=2.36.0
```
Comment (philtom-ctds) [inline requirements.txt:2]: Can `boto` be removed?  I don't imagine we want to use both `boto` and `boto3` in the same repo.

Edit:  Oh.  I see that there's `storage3.py` that live next to `storage.py`.  So, I guess this was intentional.

File: requirements.txt [hunk 2]
```diff
@@ -1,5 +1,6 @@
...
+boto3==1.9.71
```
Comment (philtom-ctds) [inline requirements.txt:3]: 1.14.17 is the current version.  Would be nice to upgrade.

File: cdisutils/storage3.py [hunk 3]
```diff
@@ -0,0 +1,668 @@
...
+                    sys.stdout.write("0.00%%\r")
+                sys.stdout.flush()
+                md5sum.update(chunk)
+                sha.update(chunk)
+                retries = 0
+
+        result['transfer_time'] = time.time() - result['start_time']
+        result['md5_sum'] = md5sum.hexdigest()
+        result['sha256_sum'] = sha.hexdigest()
+        return result
```
Comment (philtom-ctds) [inline cdisutils/storage3.py:668]: 668 lines of untested code. :(

Comment (ProfOak) [reply cdisutils/storage3.py:668]: I will file a ticket to add tests for this. It's getting to be a little outside the scope of my current task.

File: cdisutils/storage3.py [hunk 4]
```diff
@@ -0,0 +1,679 @@
...
+    """ Print the status of a transfer, given time and size """
+    size_info = get_nearest_file_size(transferred_bytes)
+    cur_time = time.clock()
+    base_transfer_rate = float(transferred_bytes) / float(cur_time - start_time)
+    transfer_info = get_nearest_file_size(base_transfer_rate)
+    cur_conv_size = float(transferred_bytes) / float(size_info[0])
+    cur_conv_rate = base_transfer_rate / float(transfer_info[0])
+    if total_size:
+        percent_complete = float(transferred_bytes) / float(total_size) * 100.0
+        sys.stdout.write(
```
Comment (singergr) [inline cdisutils/storage3.py:82]: I'm not sure `sys.stdout.write` will do exactly what we want on Python3, but that can be a problem for when we update the code consuming it to Python 3.

Comment (ProfOak) [reply cdisutils/storage3.py:82]: We could use a progress bar instead if this becomes an issue.

**Pull request merged**